Add placeholder option for each card

UTCbridget opened the pull request:
Add placeholder option for each card and if true, render the card with "none" class, meaning that particular card has no background and no shadows.

For example: If someone wants the size and position of cards to be in a three-card format, but they only provide content for one or two cards, the placeholder option preserves the  three-card format while "hiding" the unused card. 

NOTE: This will also hide any content the user may inadvertently add because it's a placeholder. If we don't add this part of the code, then any content provided will show without the background and shadows.

BEFORE (third card not hidden):

![Screen Shot 2021-08-25 at 8 26 17 AM](https://user-images.githubusercontent.com/82905787/130836440-5d31e8ca-ad8a-4558-b597-eb5c60de7805.png)

AFTER (third card hidden):

![Screen Shot 2021-08-25 at 1 14 53 PM](https://user-images.githubusercontent.com/82905787/130836457-16d1ae56-cd1e-49e3-aa0d-9d91ef35c4b5.png)

Here's is the CMS action:
(NOTE: The "Help" reads thus: "Check this if you have selected 3 cards in the "Number of Cards to Display" above and don’t want Card Three to show, but you need it as a placeholder to size and position the other cards in this series. This will hide any information you enter as well as any background and shadows from this card.")

![hide-3rd-card](https://user-images.githubusercontent.com/82905787/130841747-ad8bf08c-12ea-4ed5-8892-94b64671e40e.gif)

Also, had to remove the extra vertical space an empty card creates on the mobile. NOTE: This will be refactored to TailwindCSS when the rest of the card CSS is refactored.

![Screen Shot 2021-08-25 at 3 01 25 PM](https://user-images.githubusercontent.com/82905787/130850391-2f024150-ed17-44dd-8b77-53e1bda8210a.png)

UTCbridget committed:
Add placeholder option for each card

Add placeholder option for each card and if true, render the card with "none" class, meaning no background and no shadows.

UTCbridget committed:
Update block--utc-card-base.html.twig

Hide any information the user may have entered when designating this card as a "placeholder" only.

UTCbridget committed:
Fix spacing issues with mobile

This removes the extra space an empty card creates on the mobile. NOTE: This will be refactored to TailwindCSS when the rest of the card CSS is refactored.

UTCbridget committed:
Update _card.css

Add placeholder class to mobile to remove bottom margin.

UTCbridget committed:
Update block--utc-card.html.twig

Correct a typo on third card to added placeholder class when true

bmartinez287 commented:
Looks good, although i probably do some testing for it tomorrow just in case. However, the implementation of if statements its a great improvement. I did notice that there's an option for the first card to be set as a placeholder.
If the first card is set as a placeholder then the logic is that there's at least two cards right? else we could end up with an empty card block that's invisible but taking space.

UTCbridget commented:
Yea, I thought about not having the option of the placeholder on the first card but then thought about this scenario, which would be rare, but possible. I'm open to removing it though.

![Screen Shot 2021-08-25 at 3 16 23 PM](https://user-images.githubusercontent.com/82905787/130851552-ebde33a6-887e-4f27-86f5-c313bf9d8ed8.png)

bmartinez287 commented:
yeah, i was wondering about that one. But what happens if everything is set as hidden? or if is a card of one and they set it as hidden?

UTCbridget commented:
Nope. That's the mobile solution. Sorry. But maybe I could do something like that condition with at least two cards.

bmartinez287 commented:
Interesting, so that covers that one scenario. However that field will be visible for a card of one even if is a dummy field for it. Should that be a the bottom because of it?

UTCbridget committed:
Update block--utc-card.html.twig

If there is only one card selected, the the placeholder, if selected, is rendered null.

UTCbridget commented:
Ok, I think I've got a fix for that. I may need to add to my "Help": 

```
	{% if card_count == 1 %}
		{% set item_card_1 = {
			card_placeholder: 'Off',
		} %}
		<div class="utc-card-grid__container"> 
			{% set button_style_select = content["#block_content"].field_card_button_type.0.target_id %}
			{% set button_style = drupal_field('field_css_class', 'taxonomy_term', button_style_select)["#object"].field_css_class.value %}
			{% include 'block--utc-card-base.html.twig' with item_card_1 %}
		</div>
	{% endif %}
```

UTCbridget commented:
Also, I'm not sure I understand your last question. No fields show if placeholder is selected: 

```
{% if not card_placeholder %}
{# could use include statements for items like images and buttons in design system #}
  {% if card_link %}
    <a href="{{ card_link }}" target="{{ card_link_target }}">
  {% endif %}

  <div class="{{ card_classes }}">
      {% if (card_icon is empty) and card_image %}
         {{ card_image }}
       {% endif %}
```
...

 ```
  </div>

	{% if card_link %}
     </a>
  {% endif %}

{% else %}

  <div class="{{ card_classes }}"></div>
  
{% endif %}
```

bmartinez287 commented:
So my last question is I regard to the placement of that checkbox on a page. Mostly due to the fact that it will also show up for a card that has only one item even though it does nothing.

bmartinez287 commented:
For example if I create a card of one element and I check that box. What would happen? I think nothing will right?

UTCbridget commented:
I did change the code of the first card to where if only 1 card is selected, then the placeholder will not be applied. Would that be a better solution? I would need to put that in the "Help" text if so. Or should we leave the option available and nothing happens if selected regardless of whether they enter info or not. The user would have to trouble shoot him/herself, but it seems like an obvious fix to me. I think either solution would work and I can go either way.

New code to prevent placeholder from applying if 1 card is selected:

```
{% if card_count == 1 %}
		{% set item_card_1 = {
			card_placeholder: 'Off',
		} %}
		<div class="utc-card-grid__container"> 
			{% set button_style_select = content["#block_content"].field_card_button_type.0.target_id %}
			{% set button_style = drupal_field('field_css_class', 'taxonomy_term', button_style_select)["#object"].field_css_class.value %}
			{% include 'block--utc-card-base.html.twig' with item_card_1 %}
		</div>
	{% endif %}
```

brittrich commented:
This is a great idea/solution Bridget, all looks good to me! And I'm sure much needed. Nice work!
